### PR TITLE
cuda-kat: update recipe to Conan v2

### PR DESCRIPTION
**Pull request by valgur**
Specify library name and version:  **cuda-kat/0.2**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/cuda-kat/all/conanfile.py [hunk 1]
```diff
@@ -29,26 +33,44 @@ def _compilers_minimum_version(self):
...
+        )
 
     def package_info(self):
-        self.cpp_info.names["cmake_find_package"] = "cuda-kat"
-        self.cpp_info.names["cmake_find_package_multi"] = "cuda-kat"
+        self.cpp_info.bindirs = []
+        self.cpp_info.libdirs = []
+
+        if is_msvc(self):
+            self.cpp_info.cxxflags.append("/Zc:__cplusplus")
```
Comment (memsharded) [inline recipes/cuda-kat/all/conanfile.py:76]: This is the typical line that we would appreciate a bit more of context, like one comment explaining why this flag is necessary, in what cases it fails if it is not there. For this one, it is not very clear why the consumers would really care about the pre-processor macro, if they rely on it, it should be their build systems that activate it?

**Review by memsharded — COMMENTED** [review]
This is looking good,  many thanks for contributing it.
Just a minor comment, please have a look

File: recipes/cuda-kat/all/conanfile.py (hunk 1 above)
Comment (valgur) [reply recipes/cuda-kat/all/conanfile.py:79]: Sure. I briefly mentioned the reasoning in the commit message, but this is a much better location for it, I agree. Added it here as well.

Comment (memsharded) [reply recipes/cuda-kat/all/conanfile.py:79]: That helps a lot to simplify future maintenance, being able to read things directly in recipe, saves a lot of hops to understand things, many thanks!

**Review by memsharded — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**